lib/LaTeXML/Post.pm

```diff
@@ -218,11 +218,12 @@ sub find_documentclass_and_packages {
       $classoptions = $$entry{options} || 'onecolumn';
       $oldstyle     = $$entry{oldstyle}; }
     elsif ($$entry{package}) {
-      push(@packages, [$$entry{package}, $$entry{options} || '']); }
-  }
+      push(@packages, [$$entry{package} . ".sty", $$entry{options} || '']); } }
   if (!$class) {
     Warn('expected', 'class', undef, "No document class found; using article");
     $class = 'article'; }
+  if ($class !~ /\.cls$/) {
+    $class = $class . ".cls"; }
```

Comment:
There was also a weird oversight with the return value of this method, which returns a single list of names, but loses the distinction between classes and packages. When I tested on my bibliography, it had returned `article` as the class, and hence the preload assumed it was referring to a package, loading `article.sty`. Luckily that threw an error during loading so I noticed it. 

I think the safe way out is to append the extensions to the output list and preserve the distinction, which is what my commit does.

Reply:
it's a single list, but the class (even undef) is always first. Wouldn't it be better to fix your preload? Did you check the other callers of ```find_documentclass_and_packages```?

Reply:
Oh, there is a return convention where the first element is the class, and if no class is present the result is `undef`? Ouch. I could rework to respect that, but it feels a little 'magical' to me. Would have preferred two array refs, one with zero-or-more classes, followed by one with zero-or-more packages, to make the convention more visible (though I do know that hurts performance a little...)

I did check other callers yes, was not able to spot this convention from them on my own reading.

Reply:
Wish we had types at points like these... The existing use in MakeBibliography was using the method incorrectly (treating class and packages together), but I started from it and assumed it was correct.

So yes, I can refactor and continue using the existing method structure, by just refactoring the MakeBibliography bits. Thanks for the suggestion!

Reply:
Done!